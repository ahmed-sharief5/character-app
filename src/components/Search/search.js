import React from "react";
import PropTypes from 'prop-types';

const Search = ({ onSearch, onSubmit, byWhat }) => {
    return (
        <div style={{ display: "inline-flex"}}>
            <input 
            className="form-control mr-sm-2" 
            onChange={onSearch}
            type="search" 
            placeholder={`Search By ${byWhat}`} 
            aria-label="Search"
            />
            <button className="btn btn-outline-success my-2 my-sm-0" onClick={onSubmit} type="submit">Search</button>
        </div>
    );
}

Search.propTypes = {
    onSearch: PropTypes.func,
    onSubmit: PropTypes.func,
    byWhat: PropTypes.string
}

export default Search;