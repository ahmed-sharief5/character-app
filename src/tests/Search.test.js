import React from "react";
import Search from "../components/Search";
import renderer from "react-test-renderer";

it("renders search component correctly", () => {
    const tree = renderer.create(<Search/>).toJSON();
    expect(tree).toMatchSnapshot();
});
