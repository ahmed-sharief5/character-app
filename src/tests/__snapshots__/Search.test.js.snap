// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search component correctly 1`] = `
<div
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  <input
    aria-label="Search"
    className="form-control mr-sm-2"
    placeholder="Search By undefined"
    type="search"
  />
  <button
    className="btn btn-outline-success my-2 my-sm-0"
    type="submit"
  >
    Search
  </button>
</div>
`;
